crates/driver/src/domain/competition/mod.rs

```diff
@@ -7,24 +24,81 @@ pub use {
     auction::Auction,
     order::Order,
     quote::Quote,
-    solution::{solve, Score, Solution, SolverTimeout},
+    solution::{Score, Solution, SolverTimeout},
 };
 
-use crate::domain::eth;
+/// An ongoing competition. There is one competition going on per solver at any
+/// time. The competition stores settlements to solutions generated by the
+/// driver, and allows them to be executed as requested later. The settlements
+/// expire after a certain amount of time, at which point trying to use them
+/// will return an `[Error::SolutionNotFound]`.
+#[derive(Debug)]
+pub struct Competition {
+    pub solver: Solver,
+    pub eth: Ethereum,
+    pub simulator: Simulator,
+    pub now: time::Now,
+    pub mempools: Vec<Mempool>,
+    pub settlements: Arc<DashMap<solution::Id, settlement::Simulated>>,
```

Comment:
I was under the impression that we had decided to simplify this and only keep a single settlement per driver that can be `/settle`-ed.

I think it is also fine this way, just wanted to point that out.